include/boost/winasio/http/http_initializer.hpp

```diff
@@ -5,29 +5,56 @@
 #pragma once
 #endif // defined(_MSC_VER) && (_MSC_VER >= 1200)
 
+#include <boost/winasio/http/http_major_version.hpp>
+
 #include <http.h>
 
 namespace boost {
 namespace winasio {
 namespace http {
 
+template <HTTP_MAJOR_VERSION> 
 class http_initializer {
 public:
-  http_initializer() {
-    DWORD retCode =
-        HttpInitialize(HTTPAPI_VERSION_1,
-                       HTTP_INITIALIZE_SERVER | HTTP_INITIALIZE_CONFIG, // Flags
-                       NULL // Reserved
-        );
-    BOOST_ASSERT(retCode == NO_ERROR);
+  http_initializer() { 
+      do_init(); 
   }
+  HANDLE get_raw_http_queue() { return req_queue_; }
   ~http_initializer() {
     DWORD retCode =
         HttpTerminate(HTTP_INITIALIZE_SERVER | HTTP_INITIALIZE_CONFIG, NULL);
     BOOST_ASSERT(retCode == NO_ERROR);
   }
+
+private:
+  void do_init();
+  HANDLE req_queue_ = nullptr;
 };
 
+template <> void http_initializer<HTTP_MAJOR_VERSION::http_ver_1>::do_init() {
+  DWORD retCode =
+      HttpInitialize(HTTPAPI_VERSION_1,
+                     HTTP_INITIALIZE_SERVER | HTTP_INITIALIZE_CONFIG, // Flags
+                     nullptr // Reserved
+      );
+  BOOST_ASSERT(retCode == NO_ERROR);
+  retCode = HttpCreateHttpHandle(&req_queue_, // Req Queue
```

Comment:
The create handle should be outside the http_initializer, since one can init the http.sys, and then create 2 or more request queues. If you write like this, it is likely that you can only create one queue. Double init http.sys might work but is not a good thing.

Reply:
The HttpInitializeis the prerequisite for creating any http queue, not a one-to-one tied to HttpCreateRequestQueue. Agree.